larray/core/array.py

```diff
@@ -6031,7 +6031,12 @@ def to_excel(self, filepath=None, sheet_name=None, position='A1', overwrite_file
         sheet_name = _translate_sheet_name(sheet_name)
 
         if wide:
-            pd_obj = self.to_frame(fold_last_axis_name=True)
+            if transpose and self.ndim >= 2:
+                axes = self.axes[:-2] + Axis(self.axes[-2].labels, self.axes[-1].name) + \
+                       Axis(self.axes[-1].labels, self.axes[-2].name)
+                pd_obj = self.set_axes(axes).to_frame(fold_last_axis_name=True)
```

Comment:
Wouldn't the implementation be easier if you did it on the pandas object? Something like:

```python
pd_obj = self.to_frame(fold_last_axis_name=True)
if transpose and self.ndim >=2:
    names = df.index.names
    df.index.names = names[:-2] + ['\\'.join(reversed(names[-1].split('\\')))]
```